Sources/CoreMetrics/Metrics.swift

```diff
@@ -12,23 +12,27 @@
 //
 //===----------------------------------------------------------------------===//
 
+// MARK: Testing API
+
+internal var _enableAssertions = true
```

Comment:
not sure how I feel about this flag. can we get rid of it?

Reply:
I don't see a way to test the assertion path without it, do you have a better solution?

Reply:
dunno! @weissi @ktoso do you have any? maybe we remove the asserts?

Reply:
In NIO we test them in [`NIOCrashTester`](https://github.com/apple/swift-nio/blob/main/Sources/NIOCrashTester/CrashTests%2BEventLoop.swift).

Reply:
I used to have a big test suite for crashes but we removed that a while ago. Nowadays doing so in integration tests only. 
CrashTester sounds like the way to go -- we could "steal" that type I guess.

Reply:
yeah, go ahead. Just need to copy the three "driver" files and for every new test suite, you'll need to put an entry into this [dict](https://github.com/apple/swift-nio/blob/main/Sources/NIOCrashTester/CrashTestSuites.swift). It's able to automatically enumerate all `test*` methods, so no need to declare every test. (Here's the [method enumeration code](https://github.com/apple/swift-nio/blob/74b3a8d6a09281ea694ab49a3502012f6c5db99d/Sources/NIOCrashTester/main.swift#L76-L89))